src/main/cpp/src/regex_rewrite_utils.cu

```diff
@@ -0,0 +1,142 @@
+ * @param mr Device memory resource used to allocate the returned column's device memory.
+ * @return New BOOL column.
+ */
+template <typename BoolFunction>
+std::unique_ptr<cudf::column> literal_range_pattern_fn(cudf::strings_column_view const& strings,
+                                                       cudf::string_scalar const& prefix,
+                                                       int const range_len,
+                                                       int const start,
+                                                       int const end,
+                                                       BoolFunction pfn,
+                                                       rmm::cuda_stream_view stream,
+                                                       rmm::mr::device_memory_resource* mr)
+{
+  auto strings_count = strings.size();
+  if (strings_count == 0) return cudf::make_empty_column(cudf::type_id::BOOL8);
+
+  CUDF_EXPECTS(prefix.is_valid(stream), "Parameter prefix must be valid.");
+
+  auto d_prefix       = cudf::string_view(prefix.data(), prefix.size());
+  auto strings_column = cudf::column_device_view::create(strings.parent(), stream);
+  auto d_strings      = *strings_column;
+  // create output column
+  auto results      = make_numeric_column(cudf::data_type{cudf::type_id::BOOL8},
+                                     strings_count,
+                                     cudf::detail::copy_bitmask(strings.parent(), stream, mr),
+                                     strings.null_count(),
+                                     stream,
+                                     mr);
+  auto results_view = results->mutable_view();
+  auto d_results    = results_view.data<bool>();
+  // set the bool values by evaluating the passed function
+  thrust::transform(
+    rmm::exec_policy(stream),
+    thrust::make_counting_iterator<cudf::size_type>(0),
+    thrust::make_counting_iterator<cudf::size_type>(strings_count),
+    d_results,
+    [d_strings, pfn, d_prefix, range_len, start, end] __device__(cudf::size_type idx) {
+      if (!d_strings.is_null(idx)) {
+        return bool{
+          pfn(d_strings.element<cudf::string_view>(idx), d_prefix, range_len, start, end)};
+      }
+      return false;
+    });
+  results->set_null_count(strings.null_count());
+  return results;
+}
+
+std::unique_ptr<cudf::column> literal_range_pattern(cudf::strings_column_view const& input,
+                                                    cudf::string_scalar const& prefix,
+                                                    int const range_len,
+                                                    int const start,
+                                                    int const end,
+                                                    rmm::cuda_stream_view stream,
+                                                    rmm::mr::device_memory_resource* mr)
+{
+  auto pfn =
```

Comment:
```suggestion
  auto const pfn =
```

Reply:
In addition, it is better to define this lamba as a separate functor in an anonymous namespace:
```suggestion
namespace {
  struct literal_range_pattern_fn {
    __device__ bool operator()(...) const {
    ....
    }
    };
  }
```
The template function `literal_range_pattern_fn` above can be renamed into something like:
```
namespace {
  std::unique_ptr<cudf::column> find_literal_range_pattern(....) { ... }
}
```